apm-agent-core/src/main/java/co/elastic/apm/agent/logging/LoggingConfiguration.java

```diff
@@ -117,67 +122,94 @@ public void assertValid(Boolean value) {
         })
         .buildWithDefault(false);
 
-    public static void init(List<ConfigurationSource> sources) {
-        setLogLevel(getValue(LOG_LEVEL_KEY, sources,
-            getValue(DEPRECATED_LOG_LEVEL_KEY, sources, LogLevel.INFO.toString())));
-        setLogFileLocation(ElasticApmAgent.getAgentHome(), getValue(LOG_FILE_KEY, sources,
-            getValue(DEPRECATED_LOG_FILE_KEY, sources, DEFAULT_LOG_FILE)));
-    }
+    @SuppressWarnings("unused")
```

Comment:
That's because we don't access the configuration value through the `logFileMaxSize` `ConfigurationOption` instance. That would require the configuration registry to be initialized already. However, the registry initializes logging. In order to break up the cyclic dependency and to not accidentally initialize logging before we had the chance to configure the logging, we manually resolve these options.